src/capstone.rs

```diff
@@ -67,23 +115,196 @@ impl Capstone {
         Some(reg_name)
     }
 
-    /// Convert an instruction_id to a String naming the instruction
+    /// Converts an instruction id `insn_id` to a `String` containing the instruction name.
     pub fn insn_name(&self, insn_id: u64) -> Option<String> {
         let insn_name = unsafe {
-            let _insn_name = cs_insn_name(self.csh, insn_id as libc::c_uint);
+            let _insn_name = ffi::cs_insn_name(self.csh, insn_id as libc::c_uint);
             if _insn_name == ptr::null() {
-                return None
+                return None;
             }
 
             CStr::from_ptr(_insn_name).to_string_lossy().into_owned()
         };
 
         Some(insn_name)
     }
+
+    /// Converts a group id `group_id` to a `String` containing the group name.
+    pub fn group_name(&self, group_id: u64) -> Option<String> {
+        let group_name = unsafe {
+            let _group_name = ffi::cs_group_name(self.csh, group_id as libc::c_uint);
+            if _group_name == ptr::null() {
+                return None;
+            }
+
+            CStr::from_ptr(_group_name)
+                .to_string_lossy()
+                .into_owned()
+        };
+
+        Some(group_name)
+    }
+
+    /// Returns the current error from not enabling `CS_OPT_DETAIL`.
+    fn detail_required_error(&self) -> Option<CsErr> {
+        if *self.cs_option_state
+                .get(&CsOptType::CS_OPT_DETAIL)
+                .unwrap() == CsOptValueBool::CS_OPT_OFF as libc::size_t {
```

Comment:
Can this ever be `None`? I was a bit lazy when I originally wrote some of this, so there are panics in the library but it'd be nice to avoid introducing any more